spec/unit/policyfile_demands_spec.rb

```diff
@@ -821,24 +821,83 @@
           # This also conflicts, but only via the graphs
           "remote-cb-two" => {
             "1.2.3" => [ ]
+          },
+
+          # This has a dependency on a conflicting cookbook
+          "dep_on_conflicting_cookbook" => {
+
+            # Only the older cookbook has the conflicting dep, however we don't
+            # know how the dependency solver will solve this without doing more
+            # inspection of the graph, so the expected behavior is to error if
+            # any possible solution could have a conflict.
+            "1.0.0" => [ ["remote-cb-two", ">= 0.0.0" ] ],
+            "2.0.0" => [ ]
+
           }
 
         }
       end
 
-      context "and no explicit source is given for the conflicting cookbook" do
+      context "and the conflicting cookbook is in the run list" do
 
-        before do
-          # This is on the community site
-          policyfile.dsl.cookbook("remote-cb")
+        let(:run_list) { [ 'repo-cookbook-one', 'remote-cb', 'remote-cb-two' ] }
+
+        context "and no explicit source is given for the conflicting cookbook" do
+
+          before do
+            # This is on the community site
+            policyfile.dsl.cookbook("remote-cb")
+          end
+
+          it "raises an error describing the conflict" do
+            repo_path = File.expand_path("path/to/repo")
+
+            expected_err = <<-ERROR
+Source supermarket(https://supermarket.chef.io) and chef_repo(#{repo_path}) contain conflicting cookbooks:
```

Comment:
Minor nitpic - it would be nice if this error gave me steps I could take to try and resolve this